contributors/devel/sig-architecture/object-lifecycle.md

```diff
@@ -0,0 +1,94 @@
+The API Object Lifecycle
+=========================
+
+*This document is oriented at users who want a deeper understanding of the
+Kubernetes API structure, and developers wanting to extend the Kubernetes API.
+An introduction to using resources with kubectl can be found in [the object management overview](https://kubernetes.io/docs/concepts/overview/working-with-objects/object-management/).*
+
+**Table of Contents**
+
+<!-- toc -->
+- [States](#states)
+  - [State Machine Diagram](#state-machine-diagram)
+  - [Handling DELETING](#handling-deleting)
+<!-- /toc -->
+
+The [Kubernetes API](https://kubernetes.io/docs/api/) (and related APIs in the
+ecosystem) defines "objects" (also called resources in REST context) which are
+created, managed, and deleted over time.  Integrating with these APIs usually
+involves writing "controllers" which actuate the API object into some external
+form - for example, creating a `Pod` in the API is actuated by kubelet running
+containers on a physical or virtual machine.
+
+All Kubernetes API objects follow a common lifecycle which can be thought of as
+a state-machine, though some specific APIs extend this and offer even more
+states.  To write a correct controller, it's important to understand the common
+object lifecycle.
+
+## States
+
+All Kubernetes API objects exist in one of the following states:
+
+* `DOES_NOT_EXIST`: The object is not known to the API server.  Calling this a
+  "state" is a bit of a stretch, but it helps to be explicit.  This state does
+  not differentiate between "has not yet been created" and "has been deleted".
+* `ACTIVE`: The API server is aware of the object and the object has not been
+  deleted (the `metadata.deletionTimestamp` is not set).  While in this state,
+  any update operations (PUT, PATCH, server-side apply, etc.) will result in
+  this same state.
+* `DELETING`: The API server is aware of the object, and the object has been
+  deleted (the `metadata.deletionTimestamp` is set), and either the object has
```

Comment:
since you're specifying the operations above, maybe you should specify the operation (`DELETE`) here too, it sounds a little like one would delete an object by setting `metadata.deletionTimestamp`.

Reply:
also, you have already used "has been deleted" to mean "DOES_NOT_EXIST" ,

Reply:
Rewording, PTAL